### ISSUE TITLE
Adding `@nuxtjs/i18n` to Nuxt breaks component types (shows unknown)

### ISSUE THREAD

**Issue by martinszeltins** [nuxt/nuxt#26214]
### Environment

Stackblitz

````
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    -
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      extends
- Runtime Modules:  -
- Build Modules:    -
------------------------------
````

### Reproduction

https://stackblitz.com/github/martinszeltins/found-nuxt-components-issue?file=app.vue

### Describe the bug

When I add this module to my modules array in nuxt.config.ts then I lose all typings for my components _(for both local and 3rd party components)._ It shows that components have type of `unknown`. Before I add this module, everything works.

**nuxt.config.ts** 👇️
```diff
modules: [
+  '@nuxtjs/i18n', // This breaks all auto-import component typings (components not recognized anymore)
   '@vueuse/nuxt',
   'nuxt-primevue'
],
```

If I remove it from my modules array then everything works again.

**Screenshot before adding the module (everything works):** 👍️  👇️

![image](https://github.com/nuxt/nuxt/assets/34019878/510078e3-3e45-42d0-ab79-e678815d2325)

**Screenshot after adding the module (type unknown):** ❌  👇️

![image](https://github.com/nuxt/nuxt/assets/34019878/8cdb9168-5bdc-4642-a7a4-44829c3c5e44)

**Comment by frasza** [comment]
I know @danielroe has been inspecting and debugging this on livestream and reported some upstream bugs. Had similar issues but not sure what the status is. 🤞

**Comment by martinszeltins** [comment]
> I know @danielroe has been inspecting and debugging this on livestream and reported some upstream bugs. Had similar issues but not sure what the status is. 🤞

@frasza I didn't that, could you please link to his livestreams where he talked about this issue, I would be curious to see.

**Comment by danielroe** [comment]
Would you raise an issue upstream on `@nuxtjs/i18n`? We do need to help resolve this but I think it's not a Nuxt issue so much as a Vue typing issue.

**Comment by danielroe** [comment]
Unfortunately, Nuxt isn't in control of what a module or library may do to types with a module augmentation - module augmentations are global and outside of our scope.

**Comment by BobbieGoede** [comment]
Wasn't aware an issue was open here as well (linking for clarity https://github.com/nuxt-modules/i18n/issues/2839), I looked into this issue and it turns out it's not caused by Nuxt I18n but the combination of Primevue's types and those of Nuxt I18n. 

@danielroe 
You may recall this issue https://github.com/nuxt-modules/i18n/issues/2483, the same is happening here. These type issues arise when modules augment the `GlobalComponents` interface in `@vue/runtime-core` instead of `vue`, unfortunately the type inference breaks only after there is a mix of modules augmenting this interface in both `@vue/runtime-core` and `vue` at the same time. 

This issue should be resolved after https://github.com/primefaces/primevue/pull/5419 has been merged and released.

**Comment by danielroe** [comment]
Thank you for your investigation ❤️

**Comment by BobbieGoede** [comment]
Closing this, the fix is included in [`nuxt-primevue@3.0.0`](https://github.com/primefaces/primevue-nuxt-module/releases/tag/v3.0.0)!
